File: sample/adfa.py
#!/usr/bin/env python3

# A dummy sample program for Question 1 (1) of Homework 2b.
# Input: a pair of a DFA M and a string w
# Output: "Yes" if M accepts w. "No" otherwise.

# THIS IS A DUMMY:
# This program always outputs "Yes".

# Read a DFA M
n, t, g = map(int, input().split())
s = input()
d = []
for i in range(n):
    d.append(list(map(int, input().split())))
I = int(input())
F = list(map(int, input().split()))

assert len(s) == t
assert all(len(di) == t for di in d)
assert len(F) == g

# Read a string w
l = int(input())
w = input()

assert len(w) == l

# Always output "Yes"
print("Yes")

File: sample/adfa.js
#!/usr/bin/env node
/**
 * A dummy sample program for Question 1 (1) of Homework 2b.
 * Input: a pair of a DFA M and a string w
 * Output: "Yes" if M accepts w. "No" otherwise.
 *
 * Usage:
 *   $ node adfa.js < input_file
 *
 * THIS IS A DUMMY:
 * This program always outputs "Yes".
 */

'use strict';
const readline = require('readline');
const assert = require('assert');

// Parse the inputs
function parseInput(lines) {
  let ln = 0;  // line number

  // Read a DFA M
  let [n, t, g] = lines[ln++].split(/\s/).map(Number);
  let s = lines[ln++];
  let d = [];
  for (let i = 0; i < n; i++) {
    d.push(lines[ln++].split(/\s/).map(Number));
  }
  let I = Number(lines[ln++]);
  let F = lines[ln++].split(/\s/).map(Number);

  // Read a string w
  let l = Number(lines[ln++]);
  let w = lines[ln++];

  // Assertions
  assert(s.length == t);
  for (let i = 0; i < n; i++) {
    assert(d[i].length == t);
  }
  assert(F.length == g);
  assert(w.length == l);

  return [{
    nstate: n,
    nsymbol: t,
    nfinal: g,
    symbols: s,
    transition: d,
    initial: I,
    final: F,
  }, w];
}

// Main function
function main(input) {
  // Always output "Yes"
  console.log("Yes")
}

// Read the input and run the main function
const rl = readline.createInterface({
  input: process.stdin,
  crtlDelay: Infinity,  // to recognize '\r\n' as a single '\n'
});
const lines = [];  // buffer
rl.on('line', (line) => { lines.push(line); });
rl.on('close', () => {
  main(parseInput(lines));
});
